plugins/inlinetoolbar/plugin.js

```diff
@@ -168,13 +174,412 @@
+	 * @param {CKEDITOR.plugins.inlinetoolbar.contextDefinition} options A set of options, defining context behavior.
+	 */
+	function Context( editor, options ) {
+		/**
+		 * Editor instance.
+		 *
+		 * @property {CKEDITOR.editor}
+		 */
+		this.editor = editor;
+
+		/**
+		 * Options passed to the constructor.
+		 *
+		 * @property {CKEDITOR.plugins.inlinetoolbar.contextDefinition}
+		 */
+		this.options = options;
+
+		/**
+		 * Toolbar instance pointed by context.
+		 *
+		 * @property {CKEDITOR.ui.inlineToolbar}
+		 */
+		this.toolbar = new CKEDITOR.ui.inlineToolbar( editor );
+
+		if ( this.options && typeof this.options.priority === 'undefined' ) {
+			this.options.priority = CKEDITOR.plugins.inlinetoolbar.PRIORITY.MEDIUM;
+		}
+
+		this._loadButtons();
+	}
+
+	Context.prototype = {
+		/**
+		 * Destroys the toolbar maintained by this context.
+		 */
+		destroy: function() {
+			if ( this.toolbar ) {
+				this.toolbar.destroy();
+			}
+		},
+
+		/**
+		 * Shows the toolbar controlled by this context.
+		 *
+		 * @param {CKEDITOR.dom.element} [pointedElement] Element that should be pointed by the inline toolbar.
+		 */
+		show: function( pointedElement ) {
+			if ( pointedElement ) {
+				this.toolbar.attach( pointedElement );
+			}
+
+			this.toolbar.show();
+		},
+
+		/**
+		 * Hides the toolbar controlled by this context.
+		 */
+		hide: function() {
+			this.toolbar.hide();
+		},
+
+		/**
+		 * Performs matching against `options.refresh`.
+		 *
+		 * @private
+		 * @param {CKEDITOR.dom.elementPath} path Element path to be checked.
+		 * @param {CKEDITOR.dom.selection} selection Selection object to be passed to the `refresh` function.
+		 * @returns {CKEDITOR.dom.element/null} Returns a {@link CKEDITOR.dom.element} if matched by `options.refresh`, `null` otherwise.
+		 */
+		_matchRefresh: function( path, selection ) {
+			var ret = null;
+
+			if ( this.options.refresh ) {
+				ret = this.options.refresh( this.editor, path, selection );
+
+				if ( ret && ret instanceof CKEDITOR.dom.element === false ) {
+					ret = ( path && path.lastElement ) || this.editor.editable();
+				}
+			}
+
+			return ret;
+		},
+
+		/**
+		 * Checks if any of `options.widgets` widgets is currently focused.
+		 *
+		 * @private
+		 * @returns {CKEDITOR.dom.element/null} Returns {@link CKEDITOR.dom.element} instance that toolbar should
+		 * point to, if any matched widget is focused. `false` otherwise, meaning no tracked widget was matched.
+		 */
+		_matchWidget: function() {
+			if ( !this.options.widgets ) {
+				return null;
+			}
+
+			var widgetNames = this.options.widgets,
+				curWidgetName = this.editor.widgets && this.editor.widgets.focused && this.editor.widgets.focused.name;
+
+			if ( typeof widgetNames === 'string' ) {
+				widgetNames = widgetNames.split( ',' );
+			}
+
+			if ( CKEDITOR.tools.array.indexOf( widgetNames, curWidgetName ) !== -1 ) {
+				return this.editor.widgets.focused.element;
+			} else {
+				return null;
+			}
```

Comment:
I agree that it might add a bit to the readability. Also I have extracted `widgetNames` before the first if as it make sense to use `widgetNames` instead of `this.options.widgets` there already.